controllers/credentials_controller.go

```diff
@@ -0,0 +1,287 @@
+		r.Log.Info("peerPodsSecret has been created by the user, skipping...")
+		return ctrl.Result{}, nil
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(context.TODO(), r.Client, peerPodsSecret, func() error {
+		r.secretMapping(peerPodsSecret, ccoSecret)
+		return nil
+	}); err != nil {
+		r.Log.Info("error in creating or updating peer-pods secret", "err", err)
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	r.Log.Info("cco-secret created and mapped to peer-pods secret", "cco-secret", ccoSecret.GetName(), "peer-pods secret", peerPodsSecret.GetName())
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *SecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).Named("credentials-controller").
+		Watches(
+			&corev1.Secret{},
+			&handler.EnqueueRequestForObject{}, builder.WithPredicates(secretsFilterPredicate())).
+		Watches(&kataconfigurationv1.KataConfig{},
+			&KataConfigHandler{r}).
+		Complete(r)
+}
+
+func secretsFilterPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.ObjectNew.GetNamespace() == "openshift-sandboxed-containers-operator" && e.ObjectNew.GetName() == credentialsRequestSecretRefName
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Object.GetNamespace() == "openshift-sandboxed-containers-operator" && e.Object.GetName() == credentialsRequestSecretRefName
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			// cco-secret deletion is done by cloud-credentials-operator, followed by owned peer-pods secret deletion
+			return e.Object.GetNamespace() == "openshift-sandboxed-containers-operator" && e.Object.GetName() == peerPodsSecretName
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return false
+		},
+	}
+}
+
+// map ccoSecret fields to peer-pods compatible fields and set to peerPodsSecret
+func (r *SecretReconciler) secretMapping(peerPodsSecret *corev1.Secret, ccoSecret *corev1.Secret) {
+	ccoToPp := map[string]string{
+		"aws_access_key_id":     "AWS_ACCESS_KEY_ID",
+		"aws_secret_access_key": "AWS_SECRET_ACCESS_KEY",
+		"azure_subscription_id": "AZURE_SUBSCRIPTION_ID",
+		"azure_client_id":       "AZURE_CLIENT_ID",
+		"azure_client_secret":   "AZURE_CLIENT_SECRET",
+		"azure_tenant_id":       "AZURE_TENANT_ID",
+		"azure_region":          "AZURE_REGION",
+		"azure_resourcegroup":   "AZURE_RESOURCE_GROUP",
+	}
+
+	if peerPodsSecret.Data == nil {
+		r.Log.Info("secretMapping: peerPodsSecret data is uninitialized")
+		return
+	}
+
+	if len(ccoSecret.Data) == 0 {
+		r.Log.Info("secretMapping: ccoSecret data is uninitialized or empty")
+		return
+	}
+
+	for ccoKey, ppKey := range ccoSecret.Data {
+		if ccoToPp[ccoKey] != "" {
+			peerPodsSecret.Data[ccoToPp[ccoKey]] = ppKey
+		}
+	}
+}
+
+func (r *SecretReconciler) newOwnedPeerPodsSecret(ownedSecret *corev1.Secret) *corev1.Secret {
```

Comment:
Can you pls add some description for this method and how the lifecycle (key rotation etc) will get handled? A reference to doc will also be good